spec/system/jobseekers_can_view_a_vacancy_spec.rb

```diff
@@ -1,22 +1,34 @@
 require "rails_helper"
 
 RSpec.describe "Viewing a single published vacancy" do
-  let(:school) { create(:school) }
+  let(:school_group) { create(:local_authority) }
+  let(:school1) { create(:school) }
+  let(:school2) { create(:school) }
+  let(:vacancy) do
+    create(:vacancy, vacancy_trait, :at_multiple_schools, :with_supporting_documents, :no_tv_applications,
+           organisation_vacancies_attributes: [
+             { organisation: school1 }, { organisation: school2 }
+           ],
+           working_patterns: %w[full_time part_time],
+           subjects: %w[Physics])
+  end
 
   before do
-    vacancy.organisation_vacancies.create(organisation: school)
+    SchoolGroupMembership.find_or_create_by(school_id: school1.id, school_group_id: school_group.id)
+    SchoolGroupMembership.find_or_create_by(school_id: school2.id, school_group_id: school_group.id)
+    vacancy.reload
     visit job_path(vacancy)
   end
 
   context "when the vacancy status is published" do
-    let(:vacancy) { create(:vacancy, :published) }
+    let(:vacancy_trait) { :published }
 
     scenario "jobseekers can view the vacancy" do
       verify_vacancy_show_page_details(vacancy)
     end
 
     context "when the publish_on date is in the future" do
-      let(:vacancy) { create(:vacancy, :future_publish) }
+      let(:vacancy_trait) { :future_publish }
```

Comment:
Should all of this cleaning up stuff be put in a separate PR?

Reply:
Arguably... I'm touching this file because I had to update it with this PR anyway.

Reply:
It's more than cleaning up. I was updating this file to test that the mean geolocation is used when we create a job alert out of a similar jobs section on a job listing.

Reply:
Ah you were removing `"when creating a job alert"` and in the process you cleaned up?

Reply:
OK, now the above comment is no longer true, and I'm going to revert this file